api/pending_test_runs_medium_test.go

```diff
@@ -73,4 +73,17 @@ func TestAPIPendingTestHandler(t *testing.T) {
 		assert.Len(t, results, 1)
 		assert.Equal(t, results[0].ID, running.ID)
 	})
+
+	t.Run("/api/status/invalid", func(t *testing.T) {
+		r, _ = i.NewRequest("GET", "/api/status/invalid", nil)
+		r = mux.SetURLVars(r, map[string]string{"filter": "invalid"})
```

Comment:
What does this bit do? If it adds `?filter=invalid` the result will be `/api/status/invalid?filter=invalid`, right? Or is this mocking the behavior of the URL routing code by extracting the third component as filter?

Reply:
This is because of the way we invoke the test, which bypasses the mux middleware so we'd have to set the vars ourselves as if they were populated by the router. Next time if I need to do this again I'll make a wrapper for this.